src/Core/Grid/Column/ColumnCollection.php

```diff
@@ -90,4 +87,34 @@ public function remove($id)
 
         return $this;
     }
+
+    /**
+     * Insert new column into collection at given position
+     *
+     * @param string          $id        Existing column id
+     * @param ColumnInterface $newColumn Column to insert
+     * @param string          $position  Position: "before" or "after"
+     *
+     * @throws ColumnNotFoundException When column with gieven $id does not exist
+     */
+    private function insertByPosition($id, ColumnInterface $newColumn, $position)
+    {
+        if (!isset($this->items[$id])) {
+            throw new ColumnNotFoundException(sprintf(
+                'Cannot insert new column into collection. Column with id "%s" was not found.', $id
+            ));
+        }
+
+        $existingColumnKeyPosition = array_search($id, array_keys($this->items));
+
+        if (self::POSITION_AFTER === $position) {
+            $existingColumnKeyPosition++;
+        }
+
+        $columns = array_slice($this->items, 0, $existingColumnKeyPosition);
+        $columns = array_merge($columns, [$newColumn->getId() => $newColumn]);
+        $columns = array_merge($columns, array_slice($this->items, $existingColumnKeyPosition));
```

Comment:
You can improve performance by using this:
```php
$columns = array_slice($this->items, 0, $existingColumnKeyPosition, true) +
    [$newColumn->getId() => $newColumn] +
    array_slice($this->items, $existingColumnKeyPosition, count($this->items) - 1, true) ;
```

union `+` is eight faster than `array_merge`. More you forget to add true to preserved keys parameters
> Note that array_slice() will reorder and reset the numeric array indices by default. You can change this behaviour by setting